### DISCUSSION TITLE
Any tutorial or guide on how to use the go consequence package?

### DISCUSSION THREAD

**Discussion by Samoht1989** [USACE/go-consequences#108] [General] [answered]
Any tutorial or guide on how to use the go consequence package?

**Comment by HenryGeorgist** [comment]
No, unfortunately not right now. What is your use case? There are a few different projects that we have that leverage go-consequences, each meet different use cases, so to give the best advice we need more info on how you might want use it.

**Reply by Samoht1989** [reply]
I have depth grid rasters from HEC-RAS. How to calculate the cost associate with the damage in the watershed? We currently use HAZUS 5.0 and the inbuilt census tracts for the analysis. We are expecting similar output results from go consequence?

**Reply by HenryGeorgist** [reply]
Sure, you can do that. Are you coding in go or a different language?

**Reply by Samoht1989** [reply]
I would like to use go-consequence model interface.

**Reply by HenryGeorgist** [reply]
I can offer command line. There is no UI on go-consequences at this time.

**Reply by Samoht1989** [reply]
Yes please that helps. I'm new to the coding language and I may not have followed the last bit there. Where should I run the command line? I'm familiar with Python language.

**Reply by HenryGeorgist** [reply]
There is a command line executable that has been developed, it is compiled for linux but we could compile it for windows. You would run it via commandline by providing a path to your inventory (or using the option for using the NSI API), a path to the depth grid, and a few options on what type of grid you are providing (in your case "depth") and then what output you want.

**Reply by Samoht1989** [reply]
That would be great if you could provide me the command line executable for windows. I'm planning to download the NSI structure inventory dataset. Does it extract the required NSI structure dataset for the depth grid extent? Do we have to provide any specific county or census tract information?

**Reply by HenryGeorgist** [reply]
We have also exported go-consequences as a DLL so that you can directly import it in python to pass in the arguments. Either is possible. The process is basically that go-consequences evaluates the depth grid to determine a bounding box, it then does a spatial query on the NSI dataset (either via shapefile, geopackage, or NSI API) for that bounding box. Then the structures in the bounding box are evaluated against the depth grid one at a time. Based on your output selection, the outputs are written during (and for a moment after) the compute.

**Reply by yinluewang** [reply]
Hello. Can we have some sample data for the crop damage analysis component? I am new to go-lang, and it seems like estimating crop damage and structural damage have different setups. Do we just need a flood duration grid and crop type information? I can contribute to documentation once I know how to do it.

**Reply by HenryGeorgist** [reply]
You need a duration grid and an arrival grid. Arrival grid is used to identify when in the season a crop is impacted. Duration grid is used to define how significant the loss of the crop in terms of yeild.
You also need crop budgets, which i think i include in the base repository here. 
The crops can be downloaded via the NASS api using the library here. I think there is examples in the testing code.

**Reply by HenryGeorgist** [reply]
https://github.com/USACE/go-consequences/blob/366b2a41669589dec86ec1d98219fa6d9408ce18/cropprovider/nass-streamprovider_test.go#L45

Notice the gridded data is in projection 5070 which is the projection of the NASS CDL data. That should get you very close to a compute if you follow that pattern in the test at line 45 linked above.

**Comment by HenryGeorgist** [comment, marked as answer]
a special thanks to @Samoht1989 for writing a how to guide
https://github.com/USACE/go-consequences/wiki/How-to-use-go-consequences
